### PR TITLE
OPSAPS-51378 Set HMS and YARN cloud storage locations as CM Template Parameters

### PR DESCRIPTION
**Pull request by smaniraju**
Based on recent fixes in CM, the Cloud Storage Locations needs to be provided as template service parameters and not as safety valve parameters for the proper initialization of services like HS2.

### CODE REVIEW

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreCloudStorageServiceConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    private static final String HMS_METASTORE_DIR_TEMPLATE_PARAM = "hive_warehouse_directory";
+
+    private static final String HMS_METASTORE_EXTERNAL_DIR_TEMPLATE_PARAM = "hive_warehouse_external_directory";
+
+    @Override
+    public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
+        List<ApiClusterTemplateConfig> hmsConfigs = new ArrayList<>(2);
+        ConfigUtils.getStorageLocationForServiceProperty(source, HMS_METASTORE_DIR)
+                .map(location -> hmsConfigs.add(config(HMS_METASTORE_DIR_TEMPLATE_PARAM, location.getValue())));
```
Comment (adoroszlai) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreCloudStorageServiceConfigProvider.java]: Not a big deal™ but `map` creates a new `Optional` which is then discarded.  I think the original `ifPresent` as written in `HiveMetastoreCloudStorageRoleConfigProvider ` was both cheaper and more clear.

Comment (smaniraju) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hive/HiveMetastoreCloudStorageServiceConfigProvider.java]: Changed to ifPresent

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/yarn/YarnCloudStorageRoleConfigProvider.java [hunk 2]
```diff
@@ -15,19 +15,17 @@
...
-    private static final String NODEMANAGER_CONFIG_SAFETY_VALVE = "nodemanager_config_safety_valve";
+    private static final String NODEMANAGER_REMOTE_APP_LOG_DIR_TEMPLATE_PARAM = "yarn_nodemanager_remote_app_log_dir";
 
     private static final String NODEMANAGER_REMOTE_APP_LOG_DIR = "yarn.nodemanager.remote-app-log-dir";
 
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case YarnRoles.NODEMANAGER:
-                String cloudStorageProperty = getCloudStorageProperty(source);
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/yarn/YarnCloudStorageRoleConfigProvider.java]: Oh, actually I think the overridden `getCloudStorageProperty()` below is not needed anymore.

**Comment by lajosrodek** [comment]
Updated PR title.

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/yarn/YarnCloudStorageRoleConfigProvider.java (hunk 2 above)
Comment (smaniraju) [reply template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/yarn/YarnCloudStorageRoleConfigProvider.java]: Yes. removed the same.

**Pull request merged**